src/components/plots/areaPlot.ts

```diff
@@ -141,6 +168,22 @@ export module Plots {
 
       return { data: allPlotData.data, pixelPoints: allPlotData.pixelPoints, selection: d3.selectAll(allElements) };
     }
+
+    protected _constructAreaProjector(xProjector: _Projector, yProjector: _Projector, y0Projector: _Projector) {
+      var definedProjector = (d: any, i: number, dataset: Dataset) => {
+        var positionX = Plot._scaledAccessor(this.x())(d, i, dataset);
+        var positionY = Plot._scaledAccessor(this.y())(d, i, dataset);
+        return positionX != null && positionX === positionX &&
+               positionY != null && positionY === positionY;
+      };
+      return (datum: any, index: number, dataset: Dataset) => {
+        return d3.svg.area()
+                     .x((innerDatum, innerIndex) => xProjector(innerDatum, innerIndex, dataset))
+                     .y1((innerDatum, innerIndex) => yProjector(innerDatum, innerIndex, dataset))
+                     .y0((innerDatum, innerIndex) => y0Projector(innerDatum, innerIndex, dataset))
+                     .defined((innerDatum, innerIndex) => definedProjector(innerDatum, innerIndex, dataset))(datum, index);
```

Comment:
It might be easier to read this if we declared the D3 shape generator as a local var:
```Typescript
var areaGenerator = d3.svg.area()
                       .x((innerDatum, innerIndex) => xProjector(innerDatum, innerIndex, dataset))
                       .y((innerDatum, innerIndex) => this._getResetYFunction()(innerDatum, innerIndex, dataset))
                       .y0((innerDatum, innerIndex) => Plot._scaledAccessor(this.y0())(innerDatum, innerIndex, dataset))
                       .defined((innerDatum, innerIndex) => definedProjector(innerDatum, innerIndex, dataset));
return areaGenerator(datum, index);
```